components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/APIConsumerImplTest.java

```diff
@@ -286,4 +699,805 @@ public void renewAccessTokenTest() throws APIManagementException {
             assertTrue(true);
         }
     }
+
+    @Test
+    public void testGetSubscriptionCount() throws APIManagementException {
+        APIConsumerImpl apiConsumer = new APIConsumerImplWrapper();
+        Subscriber subscriber = new Subscriber("Subscriber");
+        apiConsumer.apiMgtDAO = apiMgtDAO;
+        when(apiMgtDAO.getSubscriptionCount((Subscriber) Mockito.anyObject(), Mockito.anyString(),
+                Mockito.anyString())).thenReturn(10);
+        apiConsumer.apiMgtDAO = apiMgtDAO;
+        assertEquals((Integer) 10, apiConsumer.getSubscriptionCount(subscriber, "testApplication",
+                "testId"));
+    }
+
+    @Test
+    public void testGetSubscribedIdentifiers() throws APIManagementException {
+        APIConsumerImpl apiConsumer = new APIConsumerImplWrapper();
+        apiConsumer.apiMgtDAO = apiMgtDAO;
+        Set<SubscribedAPI> originalSubscribedAPIs = new HashSet<SubscribedAPI>();
+        SubscribedAPI subscribedAPI = Mockito.mock(SubscribedAPI.class);
+        originalSubscribedAPIs.add(subscribedAPI);
+        Subscriber subscriber = new Subscriber("Subscriber");
+        APIIdentifier apiId1 = new APIIdentifier("admin", "API1", "1.0.0");
+        Tier tier = Mockito.mock(Tier.class);
+
+        when(apiMgtDAO.getSubscribedAPIs(subscriber, "testID")).thenReturn(originalSubscribedAPIs);
+        when(subscribedAPI.getTier()).thenReturn(tier);
+        when(tier.getName()).thenReturn("tier");
+        when(subscribedAPI.getApiId()).thenReturn(apiId1);
+        assertNotNull(apiConsumer.getSubscribedIdentifiers(subscriber, apiId1,"testID"));
+    }
+
+    @Test
+    public void testIsSubscribed() throws APIManagementException {
+        APIConsumerImpl apiConsumer = new APIConsumerImplWrapper();
+        apiConsumer.apiMgtDAO = apiMgtDAO;
+        APIIdentifier apiIdentifier = new APIIdentifier("admin", "TestAPI", "1.0.0");
```

Comment:
Can replace with constants as used in APIConsumerImplTest.java class. Do it in other such places in the PR

Reply:
if this replace with constant some test cases fail, because some test are using different identifier values and mocking apimgtDAO